packages/base-driver/lib/basedriver/commands/execute.js

```diff
@@ -0,0 +1,61 @@
+// @ts-check
+import _ from 'lodash';
+import {makeArgs, checkParams} from '../../protocol';
+
+/**
+ * @param {SessionBase} Base
+ * @returns {ExecuteBase}
+ */
+export function ExecuteMixin(Base) {
+  /**
+   * @implements {IExecuteCommands}
+   */
+  class ExecuteCommands extends Base {
+    /**
+     * @this {Driver}
+     * @param {string} script
+     * @param {[Record<string, any>]|[]} protoArgs
+     */
+    async executeMethod(script, protoArgs) {
+      // the w3c protocol will give us an array of arguments to apply to a javascript function.
+      // that's not what we're doing. we're going to look for a JS object as the first arg, so we
+      // can perform validation on it. we'll ignore everything else.
+      let args = protoArgs[0];
```

Comment:
would this works if protoArgs is undefined?

Reply:
maybe `protoArgs?.[0]` ?

Reply:
yes, good idea. i added some more validation including this case.